go/vt/vtgate/planbuilder/route.go

```diff
@@ -26,12 +26,34 @@ import (
 )
 
 // WireupRoute returns an engine primitive for the given route.
-func WireupRoute(ctx *plancontext.PlanningContext, eroute *engine.Route, sel sqlparser.SelectStatement) (engine.Primitive, error) {
-	// prepare the queries we will pass down
-	eroute.Query = sqlparser.String(sel)
-	buffer := sqlparser.NewTrackedBuffer(sqlparser.FormatImpossibleQuery)
-	node := buffer.WriteNode(sel)
-	eroute.FieldQuery = node.ParsedQuery().Query
+func WireupRoute(
+	ctx *plancontext.PlanningContext,
+	eroute *engine.Route,
+	sel sqlparser.SelectStatement,
+) (engine.Primitive, error) {
+	var transform sqlparser.NodeTransformer
+	if len(ctx.NodeTransformers) > 0 {
+		transform = sqlparser.ComposeTransformers(ctx.NodeTransformers...)
```

Comment:
I'm not quite following why we need this. Maybe we can have a chat and you can explain what you are aiming for here